pkg/controller/daemon/daemoncontroller.go

```diff
@@ -818,9 +822,8 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 			// unexpected
 			case
 				predicates.ErrPodAffinityNotMatch,
-				predicates.ErrServiceAffinityViolated,
-				predicates.ErrTaintsTolerationsNotMatch:
-				return false, false, false, fmt.Errorf("unexpected reason: GeneralPredicates should not return reason %s", reason.GetReason())
+				predicates.ErrServiceAffinityViolated:
+				return false, false, false, fmt.Errorf("unexpected reason: deamonSetPredicates should not return reason %s", reason.GetReason())
```

Comment:
Shouldn't this be logged somewhere? I see the caller check whether the error is nil, then just continue.